interbotix_ros_xseries/interbotix_xs_sdk/scripts/xs_sdk_sim.py

```diff
@@ -229,15 +229,19 @@ def robot_get_motor_configs(self) -> bool:
         mode_groups = mode_configs.get('groups', {})
         singles = mode_configs.get('singles', {})
 
-        # populate self.gripper_map
         if grippers:
             for gpr, items in grippers.items():
                 self.gripper_map[gpr] = {
-                    'horn_radius': items['horn_radius'],
-                    'arm_length': items['arm_length'],
-                    'left_finger': items['left_finger'],
-                    'right_finger': items['right_finger']
+                    'type': items.get('type', 'swing_arm'),
+                    'horn_radius': items.get('horn_radius', 0.014),
+                    'pitch_radius': items.get('pitch_radius', 0.0127),
+                    'arm_length': items.get('arm_length', 0.024),
+                    'left_finger': items.get('left_finger', 'left_finger'),
+                    'right_finger': items.get('right_finger', 'right_finger'),
+                    'calibrate': items.get('calibrate', False)
                 }
+                if (self.gripper_map[gpr]['type'] != 'swing_arm' and  self.gripper_map[gpr]['type'] != 'rack_and_pinion'):
+                    LOGGER.error((f"Invalid Gripper Type: '{self.gripper[grp]['type']}' . Options are: 'swing_arm', 'rack_and_pinion'"))
```

Comment:
Should add message on how behavior will default to act like `rack_and_pinion` if not set to either of these.

Reply:
Do we want to convert this error to fatal so that it just stops right there, than executing the default behavior?

Reply:
> convert this error to fatal

Let's do this.